std/textproto/reader.ts

```diff
@@ -0,0 +1,227 @@
+      }
+
+      //let key = canonicalMIMEHeaderKey(kv.subarray(0, endKey));
+      const key = str(kv.subarray(0, i));
+
+      // As per RFC 7230 field-name is a token,
+      // tokens consist of one or more chars.
+      // We could throw `Deno.errors.InvalidData` here,
+      // but better to be liberal in what we
+      // accept, so if we get an empty key, skip it.
+      if (key == "") {
+        continue;
+      }
+
+      // Skip initial spaces in value.
+      i++; // skip colon
+      while (
+        i < kv.byteLength &&
+        (kv[i] == charCode(" ") || kv[i] == charCode("\t"))
+      ) {
+        i++;
+      }
+      const value = str(kv.subarray(i)).replace(
+        invalidHeaderCharRegex,
+        encodeURI,
+      );
+
+      // In case of invalid header we swallow the error
+      // example: "Audio Mode" => invalid due to space in the key
+      try {
+        m.append(key, value);
+      } catch {
+        // Pass
+      }
+    }
+  }
+
+  async readLineSlice(): Promise<Uint8Array | null> {
+    // this.closeDot();
+    let line: Uint8Array | undefined;
+    while (true) {
+      const r = await this.r.readLine();
+      if (r === null) return null;
+      const { line: l, more } = r;
+
+      // Avoid the copy if the first call produced a full line.
+      if (!line && !more) {
+        // TODO(ry):
+        // This skipSpace() is definitely misplaced, but I don't know where it
+        // comes from nor how to fix it.
+        if (this.skipSpace(l) === 0) {
+          return new Uint8Array(0);
+        }
+        return l;
+      }
+      line = line ? concat(line, l) : l;
+      if (!more) {
+        break;
+      }
+    }
+    return line;
+  }
+
+  async readResponse(expectCode: number): Promise<Response> {
+    // deno-lint-ignore prefer-const
+    let [code, continued, message] = await this.readCodeLine(expectCode);
+    while (continued) {
+      const line = await this.readLine();
+      if (line === null) {
+        throw new Deno.errors.UnexpectedEof();
+      }
+      let code2: number;
+      let moreMessage: string;
+      [code2, continued, moreMessage] = await parseCodeLine(line, 0);
+      if (code2 !== code) {
+        message += "\n" + line.replace(/^\r\n/, "");
+        continued = true;
+        continue;
+      }
+      message += "\n" + moreMessage;
+    }
+    return [code, message];
+  }
+
+  skipSpace(l: Uint8Array): number {
+    let n = 0;
+    for (let i = 0; i < l.length; i++) {
+      if (l[i] === charCode(" ") || l[i] === charCode("\t")) {
+        continue;
+      }
+      n++;
+    }
+    return n;
+  }
+
+  private async readCodeLine(expectCode: number): Promise<CodeLine> {
+    const line = await this.readLine();
+    if (line === null) {
+      throw new Deno.errors.UnexpectedEof();
+    }
+    return parseCodeLine(line, expectCode);
+  }
+}
+
+function parseCodeLine(line: string, expectCode: number): CodeLine {
+  if (line.length < 4 || (line[3] !== " " && line[3] !== "-")) {
```

Comment:
A few magic numbers here.  Named constant for 3 and 4 would reveal the position significance?